OpenRA.Mods.Common/Traits/Render/WithDeployAnimation.cs

```diff
@@ -0,0 +1,86 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	[Desc("Replaces the sprite during deploy/undeploy.")]
+	public class WithDeployAnimationInfo : ITraitInfo, Requires<WithSpriteBodyInfo>
+	{
+		[Desc("Sequence name to use.")]
+		[SequenceReference] public readonly string Sequence = "make";
+
+		[Desc("Apply to sprite bodies with these names.")]
+		public readonly string[] BodyNames = { "body" };
+
+		public object Create(ActorInitializer init) { return new WithDeployAnimation(init, this); }
+	}
+
+	public class WithDeployAnimation : INotifyDeployAnimation
```

Comment:
Probably, but I didn't feel like dealing with all the building-specific BuildComplete stuff, considering that (currently) the deploy anim is only needed for actors that don't have/need the Building trait.

I'd also like to at least mention that this merge would be trivial if we got rid of all the BuildComplete hacks first (which is a large undertaking by itself, of course).